samples/WorkerBindingSamples/Blob/BlobInputBindingSamples.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+
+namespace SampleApp
+{
+    public class BlobInputBindingSamples
+    {
+        private readonly ILogger<BlobInputBindingSamples> _logger;
+
+        public BlobInputBindingSamples(ILogger<BlobInputBindingSamples> logger)
+        {
+            _logger = logger;
+        }
+
+        [Function(nameof(BlobInputClientFunction))]
+        public async Task<HttpResponseData> BlobInputClientFunction(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequestData req,
+            [BlobInput("input-container/sample1.txt", Connection = "AzureWebJobsStorage")] BlobClient client)
+        {
+            var downloadResult = await client.DownloadContentAsync();
+            var content = downloadResult.Value.Content.ToString();
+            _logger.LogInformation("Blob content: {content}", content);
+
+            return req.CreateResponse(HttpStatusCode.OK);
+        }
+
+        [Function(nameof(BlobInputStreamFunction))]
+        public HttpResponseData BlobInputStreamFunction(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequestData req,
+            [BlobInput("input-container/sample1.txt", Connection = "AzureWebJobsStorage")] Stream stream)
+        {
+            using var blobStreamReader = new StreamReader(stream);
```

Comment:
This using statement needs braces around the usage, so it's disposed at the right scope, right?

Reply:
I believe it will still get disposed within the right scope, we're using a `using declaration` here and this is what the docs say about it:

>The using declaration calls the [Dispose](https://learn.microsoft.com/en-us/dotnet/api/system.idisposable.dispose) method on the object in the correct way when it goes out of scope. The using statement causes the object itself to go out of scope as soon as [Dispose](https://learn.microsoft.com/en-us/dotnet/api/system.idisposable.dispose) is called.

>Within the using block, the object is read-only and can't be modified or reassigned. A variable declared with a using declaration is read-only. If the object implements IAsyncDisposable instead of IDisposable, either using form calls the [DisposeAsync](https://learn.microsoft.com/en-us/dotnet/api/system.iasyncdisposable.disposeasync) and awaits the returned [ValueTask](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask). For more information on [IAsyncDisposable](https://learn.microsoft.com/en-us/dotnet/api/system.iasyncdisposable), see [Implement a DisposeAsync method](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync).

If a `using statement` is a better fit here I'd like to learn more and happy to make the changes suggested

Reply:
I think you're good - I haven't actually used this new C# 8 using declaration syntax - I've always defined using blocks :)